### PR TITLE
Close client when HugeGraphStudio shutdown

### PR DESCRIPTION
**Pull request by Linary**

### CODE REVIEW

File: studio-api/src/main/java/com/baidu/hugegraph/studio/board/HugeClientWrapper.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2019 Baidu, Inc. All Rights Reserved.
```
Comment (javeme) [inline studio-api/src/main/java/com/baidu/hugegraph/studio/board/HugeClientWrapper.java]: apache 2 licence

File: studio-api/src/main/java/com/baidu/hugegraph/studio/config/StudioApiConfig.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ public String getBoardFilePath() {
     public String getBaseDirectory() {
         String userDataDir = this.config.get(StudioApiOptions.DATA_BASE_DIR);
         if (StringUtils.isBlank(userDataDir)) {
-            userDataDir = "~/.hugegraph-studio";
+            userDataDir = ".hugegraph-studio";
```
Comment (javeme) [inline studio-api/src/main/java/com/baidu/hugegraph/studio/config/StudioApiConfig.java]: add a const var

**Review by javeme — CHANGES_REQUESTED** [review]

File: studio-api/src/main/java/com/baidu/hugegraph/studio/board/service/BoardService.java [hunk 3]
```diff
@@ -273,9 +287,9 @@ public Response adjacentVertices(
         Preconditions.checkArgument(StringUtils.isNotBlank(label),
                                     "parameter label is blank");
 
-        HugeClient client = null;
+        HugeClient client;
         try {
-            client = newHugeClient();
+            client = HugeClientWrapper.get(conf);
```
Comment (javeme) [inline studio-api/src/main/java/com/baidu/hugegraph/studio/board/service/BoardService.java:292]: also pass error message to browser

File: studio-dist/src/main/java/com/baidu/hugegraph/studio/HugeGraphStudio.java [hunk 4]
```diff
@@ -61,6 +61,17 @@
     public static void main(String[] args) throws Exception {
         StudioServerConfig config = new StudioServerConfig(DEFAULT_CONFIG_FILE);
         run(config);
+        // Add shutdown hook to close studio server
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                if (server != null) {
+                    System.out.println("stop studio server");
```
Comment (javeme) [inline studio-dist/src/main/java/com/baidu/hugegraph/studio/HugeGraphStudio.java]: remove it

**Review by javeme — CHANGES_REQUESTED** [review]

File: studio-api/src/main/java/com/baidu/hugegraph/studio/board/service/BoardService.java [hunk 5]
```diff
@@ -273,12 +287,12 @@ public Response adjacentVertices(
...
         } catch (Exception e) {
             QueryResult result = new QueryResult();
-            result.setMessage("Failed to connect HugeGraphServer");
+            result.setMessage("Failed to connect HugeGraphServer" + e);
```
Comment (zhoney) [inline studio-api/src/main/java/com/baidu/hugegraph/studio/board/service/BoardService.java]: add ": " to keep consistent with line 143

**Review by zhoney — CHANGES_REQUESTED** [review]

**Review by javeme — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Pull request merged**